Assets/Scripts/Enemies/SwarmBehavior.cs

```diff
@@ -0,0 +1,135 @@
+﻿/* 
+ * Handles the health, damage and units of the Swarm as well as the Swarm interactions with the Player.
+ * 
+ * Author: Cristion Dominguez
+ * Date: 13 Nov. 2020
+ */
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+public class SwarmBehavior : MonoBehaviour
+{
+    [Header("Swarm Health and Damage")]
+    [SerializeField]
+    private int totalHealth = 100;  // health of full swarm
+
+    [SerializeField]
+    private int totalDamage = 20;  // damage done by a full swarm
+
+    [Header("Swarm Units")]
+    [SerializeField]
+    private GameObject unitPrefab;  // prefab of a swarm resident
+
+    [SerializeField]
+    private int swarmPopulation = 100;  // unit count of a full swarm
+
+    [Header("On Player Collision")]
+    [SerializeField]
+    private float disperseTime = 1.5f;
+
+    private GameObject[] units;  // arrays of units
+
+    private List<int> activeUnitIndexes;  // list of indexes corresponding to active units
```

Comment:
Is there any reason for having an Array of GameObjeccts and a list of their indexes?, from what I saw it seems like this could be simplified by just using a list of GameObjects for the Units

Reply:
The list of active unit indexes has been removed and the units array has been converted to a units list. Also updated the documentation.